### PR TITLE
29 storing files

### PR DESCRIPTION
**Pull request by hkamalo**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (edouardmisset) [inline app.js]:
```suggestion

```

File: app.js [hunk 2]
```diff
@@ -1,8 +1,11 @@
+/* eslint-disable no-unused-vars */
```
Comment (edouardmisset) [inline app.js]:
```suggestion

```

File: routes/index.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ const usersRouter = require('./users');
 const locationsRouter = require('./locations');
 const experimentsRouter = require('./experiments');
 const sensorsRouter = require('./sensors');
+const uploadRouter = require('./uploads');
```
Comment (edouardmisset) [inline routes/index.js]:
```suggestion
const storageRouter = require('./storage');
```

File: routes/index.js [hunk 4]
```diff
@@ -12,4 +13,5 @@ module.exports = (app) => {
   app.use('/locations', locationsRouter);
   app.use('/sensors', sensorsRouter);
   app.use('/experiments', experimentsRouter);
+  app.use('/upload', uploadRouter);
```
Comment (edouardmisset) [inline routes/index.js]:
```suggestion
  app.use('/storage', storageRouter);
```

File: routes/uploads.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion

```

File: routes/uploads.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-unused-vars */
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion

```

File: routes/uploads.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+/* eslint-disable import/no-extraneous-dependencies */
+const uploadRouter = require('express').Router();
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion
const storageRouter = require('express').Router();
```

File: routes/uploads.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+const multer = require('multer');
+const ExperimentModel = require('../models/ExperimentModel');
+
+const storage = multer.diskStorage({
+  // Destination to store image
+  destination: 'upload/images',
+  filename: (req, file, cb) => {
+    cb(
+      null,
+      `${file.fieldname}_${Date.now()}${path.extname(file.originalname)}`
```
Comment (edouardmisset) [inline routes/uploads.js]: Missing a dot ?
```suggestion
      `${file.fieldname}_${Date.now()}.${path.extname(file.originalname)}`
```

File: routes/uploads.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    );
+    // file.fieldname is name of the field (image)
+    // path.extname get the uploaded file extension
+  },
+});
+
+const upload = multer({ storage });
+
+uploadRouter.get('/', async (req, res) => {
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion
storageRouter.get('/', async (req, res) => {
```

File: routes/uploads.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  try {
+    const id = 2;
+    const imagePath = await ExperimentModel.selectImg(id);
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion
    const filePath = await ExperimentModel.selectImg(id);
```

File: routes/uploads.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+    return res.status(200).send(imagePath);
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion
    return res.status(200).send(filePath);
```

File: routes/uploads.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+  try {
+    const id = 2;
+    await ExperimentModel.update(id, req.file.path);
+    res.status(201).send(req.file);
+  } catch (error) {
+    return res.status(500).send(error);
+  }
+});
+
+module.exports = uploadRouter;
```
Comment (edouardmisset) [inline routes/uploads.js]:
```suggestion
module.exports = storageRouter;
```

File: models/ExperimentModel.js [hunk 13]
```diff
@@ -21,4 +21,23 @@ const create = ({
...
   });
 
-module.exports = create;
+const update = (id, path) =>
+  prisma.experiment.update({
+    where: {
+      id,
+    },
+    data: {
+      rainGraph: `${path}`,
```
Comment (edouardmisset) [inline models/ExperimentModel.js:30]: I think it should be cost graph rather than rain graph, no ?

File: models/ExperimentModel.js [hunk 14]
```diff
@@ -21,4 +21,23 @@ const create = ({
...
+    },
+  });
+
+  const selectImg = (id) => prisma.experiment.findUnique({
+    where: {
+      id
+    },
+    select: {
+      rainGraph: true,
```
Comment (edouardmisset) [inline models/ExperimentModel.js:39]: same here ?

File: routes/uploads.js (hunk 6 above)
Comment (edouardmisset) [inline routes/uploads.js]: Rename the file in storage.js and update the links to this file in order to make it more generic 😉

**Review by edouardmisset — CHANGES_REQUESTED** [review]

**Review by So-G — APPROVED** [review]
removed some dead code in app.js (line 1 & 2 referring to eslint)

**Review by edouardmisset — APPROVED** [review]

**Pull request merged**